src/marqo/index.py

```diff
@@ -119,68 +127,85 @@ def create(config: Config,
-        #     else:
-        #         cl_settings.pop('storage_class', None)
-        #     cl_settings['number_of_inferences'] = create_index_settings.number_of_inferences
-        #     cl_settings['number_of_replicas'] = create_index_settings.number_of_replicas
-        #     cl_settings['number_of_shards'] = create_index_settings.number_of_shards
-        #     response = req.post(f"indexes/{index_name}", body=cl_settings)
-        #     if create_index_settings.wait_for_readiness:
-        #         cloud_wait_for_index_status(req, index_name, IndexStatus.READY)
-        #     return response
-        return req.post(f"indexes/{index_name}", body=create_index_settings.request_body)
+
+        # py-marqo against local Marqo
+        if not config.api_key:
+            local_create_index_settings: IndexSettings = IndexSettings(
+                type=type,
+                allFields=all_fields,
+                settingsDict=settings_dict,
+                treatUrlsAndPointersAsImages=treat_urls_and_pointers_as_images,
+                shortStringLengthThreshold=short_string_length_threshold,
+                tensorFields=tensor_fields,
+                model=model,
+                modelProperties=model_properties,
+                normalizeEmbeddings=normalize_embeddings,
+                textPreprocessing=text_preprocessing,
+                imagePreprocessing=image_preprocessing,
+                vectorNumericType=vector_numeric_type,
+                annParameters=ann_parameters
+            )
+
+            return req.post(f"indexes/{index_name}", body=local_create_index_settings.generate_request_body())
+
+        # py-marqo against Marqo Cloud
+        elif config.api_key:
+            # client instantiated with api endpoint
+            if config.is_marqo_cloud:
+                cloud_index_settings: CloudIndexSettings = CloudIndexSettings(
+                    type=type,
+                    allFields=all_fields,
+                    settingsDict=settings_dict,
+                    treatUrlsAndPointersAsImages=treat_urls_and_pointers_as_images,
+                    shortStringLengthThreshold=short_string_length_threshold,
+                    tensorFields=tensor_fields,
+                    model=model,
+                    modelProperties=model_properties,
+                    normalizeEmbeddings=normalize_embeddings,
+                    textPreprocessing=text_preprocessing,
+                    imagePreprocessing=image_preprocessing,
+                    vectorNumericType=vector_numeric_type,
+                    annParameters=ann_parameters,
+                    numberOfInferences=number_of_inferences,
+                    inferenceType=inference_type,
+                    numberOfShards=number_of_shards,
+                    numberOfReplicas=number_of_replicas,
+                    storageClass=storage_class,
+                )
+            # client instantiated with endpoint url
```

Comment:
note: this should not work. it could be OK to still send this request to cloud, as the error will come back from the cloud, too. Please test it and check what the error is (the error should make sense, for example: 409 Conflict)